### PR TITLE
feat(favorites) drag-and-drop favorites section

### PR DESCRIPTION
**Pull request by gitmotion**
Fix: https://github.com/CorentinTh/it-tools/issues/316, https://github.com/CorentinTh/it-tools/issues/525

Added vuedraggable to allow dragging and dropping for the reordering of favorites
- drag and drop updates the order of the favorites
- updates the side nav favorites list with new order
- maintains the current add and remove functionality
- keeps order/state on reload
- added tooltip next to favorites label/header
- included tooltip to all locales (new update)
- added ghost-class with animation for better ux (new update)
image
demo

### CODE REVIEW

**Comment by gitmotion** [comment]
since PRs are a bit backed up you can use this image in your docker-compose file if you want this functionality along with my other additions as well until the main branch has been updated:

**docker image:**
```
ghcr.io/gitmotion/it-tools:latest
```
- you may need to clear cache and hard reload to get new features loading

#### OR locally
**github repo:**
```
git clone -b gitmotion/main https://github.com/gitmotion/it-tools.git
pnpm i && pnpm dev
``` 
- (installing packages for the first time may take some time)

File: src/pages/Home.page.vue [hunk 1]
```diff
@@ -66,10 +75,20 @@ const { t } = useI18n();
         <div v-if="toolStore.favoriteTools.length > 0">
           <h3 class="mb-5px mt-25px font-500 text-neutral-400">
             {{ $t('home.categories.favoriteTools') }}
+            <c-tooltip tooltip="Drag and drop to reorder favorites">
```
Comment (CorentinTh) [inline src/pages/Home.page.vue:78]: Put this text line in en locale file please 
```suggestion
            <c-tooltip :tooltip="$t('...')">
```

**Review by CorentinTh — APPROVED** [review]
Good idea, thanks :pray:

File: src/pages/Home.page.vue [hunk 2]
```diff
@@ -66,10 +75,20 @@ const { t } = useI18n();
...
           </h3>
-          <div class="grid grid-cols-1 gap-12px lg:grid-cols-3 md:grid-cols-3 sm:grid-cols-2 xl:grid-cols-4">
-            <ToolCard v-for="tool in toolStore.favoriteTools" :key="tool.name" :tool="tool" />
-          </div>
+          <Draggable
+            :list="favoriteTools"
+            class="grid grid-cols-1 gap-12px lg:grid-cols-3 md:grid-cols-3 sm:grid-cols-2 xl:grid-cols-4"
+            item-key="name"
+            @end="onUpdateFavoriteTools"
+          >
```
Comment (CorentinTh) [inline src/pages/Home.page.vue:87]: Maybe also add a gost class for better ux

```suggestion
          <Draggable
            :list="favoriteTools"
            class="grid grid-cols-1 gap-12px lg:grid-cols-3 md:grid-cols-3 sm:grid-cols-2 xl:grid-cols-4"
            item-key="name"
            @end="onUpdateFavoriteTools"
            ghost-class="..."
          >
```

**Review by CorentinTh — CHANGES_REQUESTED** [review]

File: src/pages/Home.page.vue (hunk 1 above)
Comment (gitmotion) [reply src/pages/Home.page.vue]: good point 👍, rebased changes to original commit to include:
- ```<c-tooltip :tooltip="$t('home.categories.favoritesDndToolTip')">```
- updated all available locales as well

File: src/pages/Home.page.vue (hunk 2 above)
Comment (gitmotion) [reply src/pages/Home.page.vue]: nice idea - updated to include the ghost-class:
demo

**Comment by gitmotion** [comment]
@CorentinTh comments resolved 👍

**Review by CorentinTh — APPROVED** [review]
Nice, thank you :pray:

**Comment by gitmotion** [comment]
> Nice, thank you :pray: 

thanks for all the feedback on this one 🙏🏼

**Pull request merged**